tools/generator/game.py

```diff
@@ -44,11 +44,27 @@ def load(self):
         self.game['user_function'] = self.game.get('user_function', [])
 
         validate_schema(self.game, GAME_SCHEMA)
+        self.load_constants()
         self.load_base_types()
         self.load_used_types()
         self.load_funcs()
         self.check()
 
+    def load_constants(self):
+        '''Load the constants and infer their type if necessary.'''
+        cst_type_mapping = {
+            int: 'int',
+            str: 'string',
+            float: 'double',
+        }
+        for cst in self.game['constant']:
+            val_type = type(cst['cst_val'])
+            if val_type not in cst_type_mapping:
+                raise GameError("Constant '{}' has unsupported type: {}"
+                                .format(cst['cst_name'], val_type))
+            if 'cst_type' not in cst:
+                cst['cst_type'] = cst_type_mapping[val_type]
```

Comment:
Maybe add a check, `elif cst['cst_type'] != cst_type_mapping[val_type]: raise …`

Reply:
I thought about doing that, but I think we should maybe just deprecate the cst_type field in the yaml and ignore it completely

Reply:
The more reason to add at least a warning if the detected type is incorrect. If the user think its `2` will be interpreted as a float…